vdr/api/v1/api.go

```diff
@@ -79,9 +80,47 @@ func (a *Wrapper) Routes(router core.EchoRouter) {
 
 // CreateDID creates a new DID Document and returns it.
 func (a Wrapper) CreateDID(ctx echo.Context) error {
-	doc, err := a.VDR.Create()
+	req := DIDCreateRequest{}
+	if err := ctx.Bind(&req); err != nil {
+		return ctx.String(http.StatusBadRequest, fmt.Sprintf("create request could not be parsed: %s", err.Error()))
+	}
+
+	options := doc2.DefaultCreationOptions()
+	if req.Controllers != nil {
+		for _, c := range *req.Controllers {
+			id, err := did.ParseDID(c)
+			if err != nil {
+				return ctx.String(http.StatusBadRequest, fmt.Sprintf("controller entry (%s) could not be parsed: %s", c, err.Error()))
+			}
+			options.Controllers = append(options.Controllers, *id)
+		}
+	}
+
+	if req.Authentication != nil && *req.Authentication {
```

Comment:
The functioning of this is now dependant on what the defaults are which is dangerous since defaults may change (and we forget to change these if's). Wouldn't it be both simpler and more adaptive to just do;

```
if req.Authentication != nil {
    options.Authentication = *req.Authentication
}
```

Since if it's set, you can override the default value.

Same ofc goes for the `if`s below.